src/software/ai/intent/intent.h

```diff
@@ -90,17 +116,59 @@ class Intent
      */
     std::set<MotionConstraint> getMotionConstraints(void) const;
 
+    /**
+     * Gets the navigator params if this intent requires navigation. For example,
+     * StopIntent does not need to navigate, while MoveIntent does.
+     *
+     * @return navigator params
+     */
+    std::optional<NavigatorParams> getNavigatorParams() const;
+
+    /**
+     * Get PrimitiveMsg that underlies this Intent
+     *
+     * @return PrimitiveMsg
+     */
+    PrimitiveMsg getPrimitiveMsg() const;
+
+    /**
+     * Gets a PrimitiveMsg updated with destination and final_speed
+     * NOTE: not all Intents update PrimitiveMsg
+     *
+     * @param destination The destination
+     * @param final_speed The final speed
+     *
+     * @return PrimitiveMsg updated with destination and final_speed
+     */
+    virtual PrimitiveMsg getUpdatedPrimitiveMsg(Point destination,
```

Comment:
hm, this get overridden by every intent that wants to navigate. I'm not sure how we can integrate this into navigator

Reply:
Sorry I wasn't clear here, wasn't saying we should integrate this into the Navigator

We should limit the scope of this function to just the navigator by making it protected, and then making the Navigator a friend of this class, so that it can access that function.

Its a really good example of when you should use `friend` imo, the primitive msg tracked inside is internal to the intent, and only the navigator cares about updating it.

Reply:
Yeah that's fair, so then we want intent test classes be friends of Intent too?